lib/cache-key.js

```diff
@@ -0,0 +1,8 @@
+'use strict';
+var crypto = require('crypto');
+
+module.exports = function cacheKey(name, dir, _hashTreeOverride) {
+  var value = name + 0x00 + dir + 0x00 + (typeof _hashTreeOverride === 'function');
```

Comment:
what's the point of including the type of `_hashTreeOverride` in the key when `skipCache` is implicitly set in the case `_hashTreeOverride` is a function?